languages/ruby/ruby_on_rails.md

```diff
@@ -3,34 +3,38 @@
 
 ## READMEs
-Please delete the Rails boilerplate one when starting a new app and start plumbing it with the what’s-where basics.  This page is what’s shown on the Github homepage for the project and is a good launching point to other pages on the wiki. Use Markdown when appropriate to improve formatting.
+Please delete the Rails boilerplate one when starting a new app and start plumbing it with the what’s-where basics.  This page is what’s shown on the Github homepage for the project and is a good launching point to other pages on the wiki. Use Markdown when appropriate to improve formatting.  Always aim for a balance between "enough to handle common tasks" and "too long to effectively find what I need".
 
-## Github Wiki
-Anything beyond the über-basics should be documented on the Github wiki pages.  Some examples of this might be “Deployment Anatomy,” “Search Architecture,” “Importing + Refreshing Application Content,” etc…
+## Wiki's
+This can be a Github wiki or whatever tool is commonly used in the organization.  Anything beyond the über-basics should be documented on these technical wiki pages.  Some examples of this might be “Deployment Anatomy,” “Search Architecture,” “Importing + Refreshing Application Content,” etc…  
 
 ## Code Comments
-Generally these are to be avoided only because they tend to be out of date if used too heavily.  You should favor using descriptive variable and method names.  
+Generally these are to be avoided only because they tend to be out of date if used too heavily.  You should favor using **descriptive variable and method names**.  
 Rather, please use inline code comments when:
 * The following lines of code might use some syntax-fu (ie. anything overly-clever)
 * You’re working around a bug in a 3rd party library/api/model that we can’t control, i.e. when we have to write something that seems bad in order to use an external/immutable dependency.
 * Magic numbers: Explain your units and why the number exists.
-* Class documentation: Consider documenting the controller or model to give the big picture so that folks can find their bearings.
+* Class documentation: Consider documenting service and library classes.  They often can interact together in ways that might not be immediately obvious from just the names.  Model classes that are especially meaty can also benefit from documentation, especially as new team members come on-board.
 * In the Gemfile when locking a gem to a specific version: Please document why we must use that version. This makes it easier to know when it is safe/reasonable to upgrade the gem.
```

Comment:
I guess this ties in with above comment on more in-depth `readmes`

Reply:
Actually... not necessarily.  The Class documentation being discussed here is in code descriptions.  There's actually a rubocop rule for this **and its enabled by default**.  
https://docs.rubocop.org/rubocop/cops_style.html#styledocumentation

The rubocop.yml in this repo is very out of date so I've been working on a new one.  (separate PR sometime soon)  A *lot* of rules have been added since then and many of them are now enabled by default.  My config for that rule currently looks like:

```rb
Style/Documentation: # MVC & Rails conventions are *often* sufficient documentation for these classes
  Exclude:
    - app/mailers/**/*
    - app/models/**/*
    - app/components/**/*
    - app/controllers/**/*
    - app/helpers/**/*
    - config/**/*
    - db/**/*
```

Would love to hear if others disagree with this approach or not.  I've left several directories in the "include" list intentionally: `services`, `lib`.  I'd like to have a way to include `app/controllers/concerns` since it may be a bit less obvious from just seeing routing and controller methods.

Reply:
I like the idea of buffing out the rubocop / code styling as a separate thing. It definitely warrants proper attention.